In [1]:
#Importing the packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
#Importing the dataset
df = pd.read_excel('ANZ synthesised transaction dataset.xlsx')

## Basic data wrangling

In [3]:
list(df)

['status',
 'card_present_flag',
 'bpay_biller_code',
 'account',
 'currency',
 'long_lat',
 'txn_description',
 'merchant_id',
 'merchant_code',
 'first_name',
 'balance',
 'date',
 'gender',
 'age',
 'merchant_suburb',
 'merchant_state',
 'extraction',
 'amount',
 'transaction_id',
 'country',
 'customer_id',
 'merchant_long_lat',
 'movement',
 'Month']

In [4]:
#Checking the datatypes
df.dtypes

status                       object
card_present_flag           float64
bpay_biller_code             object
account                      object
currency                     object
long_lat                     object
txn_description              object
merchant_id                  object
merchant_code               float64
first_name                   object
balance                     float64
date                 datetime64[ns]
gender                       object
age                           int64
merchant_suburb              object
merchant_state               object
extraction                   object
amount                      float64
transaction_id               object
country                      object
customer_id                  object
merchant_long_lat            object
movement                     object
Month                         int64
dtype: object

In [5]:
df.sample()

status  card_present_flag bpay_biller_code         account currency  \
7489  authorized                1.0              NaN  ACC-2673069055      AUD   

           long_lat txn_description                           merchant_id  \
7489  152.99 -27.49       SALES-POS  df271e23-5dfa-4d26-9a47-ed2441cdee3a   

      merchant_code first_name  ...  merchant_suburb merchant_state  \
7489            NaN    Richard  ...   Toowoomba City            QLD   

                        extraction  amount                    transaction_id  \
7489  2018-09-27T16:21:03.000+0000    7.71  fd1fba2ee85f4971a06425050365335f   

        country   customer_id  merchant_long_lat movement Month  
7489  Australia  CUS-51506836      151.95 -27.56    debit     9  

[1 rows x 24 columns]

In [6]:
#Checking for duplicates
sum(df.duplicated())

0

In [7]:
#Checking for missing data
df.isnull().sum()

status                   0
card_present_flag     4326
bpay_biller_code     11158
account                  0
currency                 0
long_lat                 0
txn_description          0
merchant_id           4326
merchant_code        11160
first_name               0
balance                  0
date                     0
gender                   0
age                      0
merchant_suburb       4326
merchant_state        4326
extraction               0
amount                   0
transaction_id           0
country                  0
customer_id              0
merchant_long_lat     4326
movement                 0
Month                    0
dtype: int64

 - There are several columns having missing values in our dataset. However, the columns 'merchant_code' and 'bpay_biller_code' have almost 93% rows as missing values. Thus, these columns cannot be used for any significant analysis. Thus, we are dropping these columns. 

 - The 'merchant_state' and 'merchant_suburb' columns also have almost 40% values missing. 

In [8]:
#Dropping 'merchant_code' and 'bpay_biller_code' columns from our dataset.
df.drop(['merchant_code','bpay_biller_code'],axis=1,inplace = True)

In [9]:
#Getting dimensions of the df
df.shape

(12043, 22)

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12043 entries, 0 to 12042
Data columns (total 22 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   status             12043 non-null  object        
 1   card_present_flag  7717 non-null   float64       
 2   account            12043 non-null  object        
 3   currency           12043 non-null  object        
 4   long_lat           12043 non-null  object        
 5   txn_description    12043 non-null  object        
 6   merchant_id        7717 non-null   object        
 7   first_name         12043 non-null  object        
 8   balance            12043 non-null  float64       
 9   date               12043 non-null  datetime64[ns]
 10  gender             12043 non-null  object        
 11  age                12043 non-null  int64         
 12  merchant_suburb    7717 non-null   object        
 13  merchant_state     7717 non-null   object        
 14  extrac

In [11]:
df.describe()

card_present_flag        balance           age        amount  \
count        7717.000000   12043.000000  12043.000000  12043.000000   
mean            0.802644   14704.195553     30.582330    187.933588   
std             0.398029   31503.722652     10.046343    592.599934   
min             0.000000       0.240000     18.000000      0.100000   
25%             1.000000    3158.585000     22.000000     16.000000   
50%             1.000000    6432.010000     28.000000     29.000000   
75%             1.000000   12465.945000     38.000000     53.655000   
max             1.000000  267128.520000     78.000000   8835.980000   

              Month  
count  12043.000000  
mean       9.011957  
std        0.816511  
min        8.000000  
25%        8.000000  
50%        9.000000  
75%       10.000000  
max       10.000000

## Data Analysis

In [12]:
#Making a copy of the df
trans = df.copy()

In [13]:
trans.groupby(['account'])['amount'].sum().sort_values()

account
ACC-2988263882    10385.54
ACC-559365433     10587.42
ACC-1217063613    10652.72
ACC-721712940     10845.25
ACC-53508546      11438.37
                    ...   
ACC-240804743     36786.13
ACC-2231965366    37943.79
ACC-1523339231    40215.54
ACC-1496451953    42688.30
ACC-3317636250    45409.16
Name: amount, Length: 100, dtype: float64

In [14]:
#Average trans amount per account
trans.groupby(['account'])['amount'].describe()

count        mean          std   min      25%     50%  \
account                                                                 
ACC-1037050564  259.0   74.465019   190.426861  3.71  14.1000  25.380   
ACC-1056639002   86.0  265.134767   808.951487  5.97  15.3200  35.170   
ACC-1199531521   77.0  418.742468  1096.170654  6.31  20.0000  28.670   
ACC-1217063613   25.0  426.108800  1336.447627  7.06  14.7300  20.000   
ACC-1222300524  303.0   78.206106   245.088217  2.10   8.6900  15.970   
...               ...         ...          ...   ...      ...     ...   
ACC-854938045   169.0  156.932249   500.109586  5.62  25.2500  41.180   
ACC-90814749    124.0  167.893548   513.971890  6.90  21.4850  36.000   
ACC-958000567   174.0  210.277299   700.277440  5.96  17.7125  31.875   
ACC-964839203   118.0  222.325932   505.626889  3.52  23.7725  31.545   
ACC-966140392   152.0  154.711842   537.308981  5.92  12.8450  21.560   

                    75%      max  
account                           
ACC-1037050564  41.7150   892.09  
ACC-1056639002  54.8100  3195.01  
ACC-1199531521  93.0000  3785.78  
ACC-1217063613  41.7500  4863.62  
ACC-1222300524  26.7100  2349.55  
...                 ...      ...  
ACC-854938045   58.0000  2538.68  
ACC-90814749    57.2025  2308.67  
ACC-958000567   59.9200  3517.61  
ACC-964839203   52.6250  1956.00  
ACC-966140392   35.0000  2572.91  

[100 rows x 8 columns]

In [15]:
trans.groupby(['gender'])['amount'].sum()

gender
F     970322.63
M    1292961.57
Name: amount, dtype: float64

In [16]:
#Grouping transactions by gender
trans.groupby(['gender'])['amount'].describe()

count        mean         std  min    25%    50%     75%      max
gender                                                                    
F       5758.0  168.517303  525.177524  0.1  15.32  27.58  51.385  7081.09
M       6285.0  205.721809  647.774391  0.1  16.68  30.00  55.780  8835.98

In [17]:
#Grouping transactions by states
trans.groupby(['merchant_state'])['amount'].describe()

count       mean         std   min      25%     50%      75%  \
merchant_state                                                                  
ACT               73.0  66.803836  201.821548  4.50  10.9500  22.910  33.4700   
NSW             2169.0  47.036316  146.858464  0.10  13.3100  22.530  40.7900   
NT               205.0  44.726293  138.170485  1.71  12.5200  17.810  31.1200   
QLD             1556.0  34.372397  218.024779  0.10   9.2875  14.700  25.8800   
SA               415.0  40.425470   87.115715  3.28  14.1500  22.120  35.9050   
TAS               68.0  28.866618   26.782770  7.35  13.7150  22.080  29.0700   
VIC             2131.0  41.099953  130.790309  0.10  14.0350  21.730  35.4750   
WA              1100.0  30.901873   72.522139  2.05  11.4300  18.365  28.7725   

                    max  
merchant_state           
ACT             1348.47  
NSW             4233.00  
NT              1285.09  
QLD             7081.09  
SA               909.61  
TAS              133.31  
VIC             3680.71  
WA              1692.56

In [18]:
trans.groupby(['merchant_state'])['amount'].sum().sort_values()

merchant_state
TAS      1962.93
ACT      4876.68
NT       9168.89
SA      16776.57
WA      33992.06
QLD     53483.45
VIC     87584.00
NSW    102021.77
Name: amount, dtype: float64

 - We have extracted the month of transaction using MS Excel.

In [19]:
trans.groupby(['Month'])['amount'].describe()

count        mean         std   min     25%    50%     75%      max
Month                                                                      
8      3943.0  185.121867  579.376003  1.52  15.455  28.13  51.425  8835.98
9      4013.0  182.045903  574.694422  0.10  16.090  29.00  54.400  8835.98
10     4087.0  196.427323  621.857821  0.10  16.390  29.00  55.000  8835.98

In [20]:
# Grouping by type of card used
trans.groupby(['movement'])['amount'].describe()

count         mean          std    min      25%      50%      75%  \
movement                                                                        
credit      883.0  1898.728029  1150.364621  576.0  1013.67  1626.48  2538.68   
debit     11160.0    52.572343   156.354143    0.1    15.19    26.93    45.00   

              max  
movement           
credit    8835.98  
debit     7081.09

In [21]:
trans.groupby(['movement'])['amount'].sum()

movement
credit    1676576.85
debit      586707.35
Name: amount, dtype: float64

In [22]:
#Dividing the customers into 4 categories based on their ages
bin_names=['Student','Junior','Senior','Retired']
bin_edges=(18,25,40,60,99)
trans['life_stage'] = pd.cut(trans['age'],bin_edges,labels=bin_names)

### Life-Stage column key(Age):
 - 18 -> 25 : Student
 - 26 -> 40 : Junior
 - 41 -> 60 : Senior
 - 61 -> 99 : Retired

In [23]:
trans.groupby(['life_stage'])['amount'].describe()

count        mean         std   min      25%    50%       75%  \
life_stage                                                                   
Student     4146.0  180.116331  573.439444  0.10  14.8025  27.00   46.7825   
Junior      5788.0  181.996579  613.318361  0.10  15.3775  29.00   55.0000   
Senior      1443.0  238.300873  593.603847  2.93  21.7100  34.66   87.8250   
Retired      224.0  179.075402  303.967860  5.10  19.2600  37.16  147.0000   

                max  
life_stage           
Student     6107.23  
Junior      8835.98  
Senior      3785.78  
Retired     1043.12

In [24]:
trans.groupby(['life_stage'])['amount'].sum()

life_stage
Student     746762.31
Junior     1053396.20
Senior      343868.16
Retired      40112.89
Name: amount, dtype: float64

In [33]:
#Grouping all the transactions by their suburbs
trans.groupby(['merchant_suburb'])['amount'].describe().sort_values(by = 'count')

count       mean         std    min     25%    50%     75%  \
merchant_suburb                                                                
Pearsall            1.0  28.780000         NaN  28.78  28.780  28.78  28.780   
Darra               1.0  19.290000         NaN  19.29  19.290  19.29  19.290   
Pacific Paradise    1.0  14.490000         NaN  14.49  14.490  14.49  14.490   
Ormond              1.0   8.890000         NaN   8.89   8.890   8.89   8.890   
Deagon              1.0  18.570000         NaN  18.57  18.570  18.57  18.570   
...                 ...        ...         ...    ...     ...    ...     ...   
Chatswood          55.0  33.833455   38.522740   4.76  11.355  18.50  43.115   
Brisbane City      79.0  34.486709   43.990472   2.40  10.020  17.58  33.460   
Southport          82.0  22.217073   58.169849   2.63   6.300   9.95  16.445   
Sydney            233.0  93.027639  146.252814   4.87  21.000  47.16  98.430   
Melbourne         255.0  46.837373  231.764098   0.10  12.330  18.94  39.615   

                      max  
merchant_suburb            
Pearsall            28.78  
Darra               19.29  
Pacific Paradise    14.49  
Ormond               8.89  
Deagon              18.57  
...                   ...  
Chatswood          196.45  
Brisbane City      220.38  
Southport          475.52  
Sydney            1233.80  
Melbourne         3680.71  

[1609 rows x 8 columns]

## Findings
 - Average amount spent per transaction is 187.93 AUD.
 - Average balance amount is 14704.195 AUD.
 - The age of customers ranges between 18 and 78.
 - The lowest and highest total spend by an account are 10385.54 and 45409.16 respectively.
 - The number of transactions made and the average spend per transactions made by male customers is more than female cutomers.
 - The sum of all the transactions made by male customers is also significantly greater than those made by female customers.
 - Highest number of transactions have been made in New South Wales and Victoria regions.
 - The Austrailian Capital Territory has the highest average spend per transaction, but the number of transactions is very very low.
 - Lowest number of transactions have been made in ACT and Tasmania states/regions.
 - New South Wales has the highest total spend.
 - Highest number of transactions were done in the month of October, the average spend per transaction is also highest in the month of October.
 - Average number of transactions made per month is 4014.33 . 
 - Number of transactions made using debit cards(11160) is significantly higher than those made using credit cards(833) .
 - Only 7.3% of the transactions have been performed using credit card but they amount for 74% of the total money spent.
 - However, the average amount spent per transaction using credit cards (1898.72) is significantly higher than that for debit cards(52.5) .
 - The total amount spent on credit card transactions is almost thrice the amount spent on debit card transactions.
 - Generally, people use credit cards for high-end or expensive transactions.
 - Highest number of transactions have been carried out by customers between the ages 18 and 40(86%) .
 - The transactions made by the above mentioned ages are worth 80% of the money spent by all the customers.
 - Melbourne and Sydney are the most active suburbs in terms of number of transactions.